staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -245,17 +251,26 @@ func (c *delegatingCRConverter) ConvertToVersion(in runtime.Object, target runti
 		return converted, nil
 	}
 
-	convertedObjects, err := c.converter.Convert(list, toGVK.GroupVersion())
+	// Deepcopy because the converter might mutate the objects.
+	for i := range objectsToConvert {
+		objectsToConvert[i] = *objectsToConvert[i].DeepCopy()
+	}
+
+	// Do the (potentially mutating) conversion.
+	convertedObjects, err := c.converter.Convert(&unstructured.UnstructuredList{
+		Object: list.Object,
+		Items:  objectsToConvert,
+	}, toGVK.GroupVersion())
```

Comment:
@liggitt with lots of assumptions around the CRConverter interface we could allow mutation and make use of it by not deepcopying furtherdown. Not sure it is worth it.